File: engine/apps/proxy-service/src/schemas/scenarios.py
"""Schemas for attack scenario catalogues."""

from __future__ import annotations

from pydantic import BaseModel


class ScenarioItem(BaseModel):
    label: str
    prompt: str
    tags: list[str]
    expectedDecision: str  # noqa: N815  — matches frontend convention


class ScenarioGroup(BaseModel):
    label: str
    color: str
    icon: str
    items: list[ScenarioItem]

File: engine/apps/frontend/app/types/scenarios.ts
export interface ScenarioItem {
  label: string
  prompt: string
  tags: string[]
  expectedDecision: 'BLOCK' | 'MODIFY' | 'ALLOW'
}

export interface ScenarioGroup {
  label: string
  color: string
  icon: string
  items: ScenarioItem[]
}
